src/CommandingOfficers/CommanderAve.java

```diff
@@ -260,27 +257,42 @@ private void relevelSnow(GameMap gameMap, GameEventQueue outEvents)
 
         // This will hold the tiles we are spreading snow to.
         HashSet<XYCoord> shallowTiles = new HashSet<XYCoord>();
-        while( (snowToSpread > 0) && (!workingSet.isEmpty()) )
+        while( (snowToSpread > 0) )
         {
           log("Snow to spread: " + snowToSpread);
           // Pull out the most empty tiles still in the working set.
-          double mostShallow = workingSet.peek().snowDepth;
-          log(" most shallow: " + mostShallow);
-          while( !workingSet.isEmpty() && (workingSet.peek().snowDepth == mostShallow) )
+          double mostShallow = 0;
+          double nextMostShallow = 0;
+          if( !workingSet.isEmpty() )
+          {
+            mostShallow = workingSet.peek().snowDepth;
+            while( !workingSet.isEmpty() && (workingSet.peek().snowDepth == mostShallow) )
+            {
+              log(" adding " + workingSet.peek().snowCoord + " to working set");
+              shallowTiles.add(workingSet.poll().snowCoord);
+            }
+            nextMostShallow = (workingSet.isEmpty()) ? mostShallow+snowToSpread : workingSet.peek().snowDepth;
+            if( nextMostShallow > MAX_SNOW_DEPTH ) nextMostShallow = MAX_SNOW_DEPTH; // Keep snow under the limit.
+          }
+          else
           {
-            log(" adding " + workingSet.peek().snowCoord + " to working set");
-            shallowTiles.add(workingSet.poll().snowCoord);
+            mostShallow = snowMap[leaf.xCoord][leaf.yCoord];
+            if( mostShallow == MAX_SNOW_DEPTH )
+              shallowTiles.remove(leaf); // If leaf is max depth, just push extra to other unprocessed tiles.
+            nextMostShallow = mostShallow + snowToSpread;
           }
```

Comment:
I don't understand what this else branch is supposed to do. The rest of this section only seems to spread snow if shallowTiles has something in it.

Reply:
Correct. But `shallowTiles` can have more than one thing in it. If we get to the point where `leaf` and its neighbors are all at `MAX_SNOW_DEPTH`, then we don't want to add any more snow to `leaf`, so we remove it from `shallowTiles` and adjust `nextMostShallow` so that the remaining snow will be evenly spread across the (as-yet unprocessed) adjacent nodes.
This statement can also be hit when we aren't yet at `MAX_SNOW_DEPTH` - in this case we don't remove `leaf` before the final snow-spreading.

Reply:
Oh, and if there are no adjacent unprocessed tiles (i.e. if we are running into a corner), then removing `leaf` will cause `shallowTiles` to be empty, and any remaining snow just vanishes in a puff of logic.

Reply:
Okay, I see what you're saying.
It's now clear to me that I just don't understand exactly how this all works, and my brain is too addled at this point to fix that.